File: src/py-opentimelineio/opentimelineio/schema/stream_address.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.StreamAddress)
def __str__(self):
    return "StreamAddress()"


@add_method(_otio.StreamAddress)
def __repr__(self):
    return "otio.schema.StreamAddress()"

File: src/opentimelineio/streamAddress.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/streamAddress.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION_NS {

StreamAddress::StreamAddress()
    : Parent()
{}

StreamAddress::~StreamAddress()
{}

bool
StreamAddress::read_from(Reader& reader)
{
    return Parent::read_from(reader);
}

void
StreamAddress::write_to(Writer& writer) const
{
    Parent::write_to(writer);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION_NS
